src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class OperationExtensions
+    {
+        /// <summary>
+        /// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+        /// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+        /// </summary>
+        public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+        {
+            /*
+            |    code         | Read | Write | ReadableRef | WritableRef |
+            | nameof(x)       |      |       |             |             |
+            | x.Prop = 1      |      |  ✔️   |             |             |
+            | x.Prop += 1     |  ✔️  |  ✔️   |             |             |
+            | x.Prop++        |  ✔️  |  ✔️   |             |             |
+            | Foo(x.Prop)     |  ✔️  |       |             |             |
+            | Foo(x.Prop)*    |      |       |     ✔️      |             |
+            | Foo(out x.Prop) |      |       |             |     ✔️      |
+            | Foo(ref x.Prop) |      |       |     ✔️      |     ✔️      |
+
+            * where void Foo(in T v)
```

Comment:
oh, i see.  you had an aseterix to refer to this line.  that's not at *all* clear.  That's because `* ` is common in multiline comments to just indicate a left margin.  Please either use `#if false` here, or ``//` comemnts, or also add a `(see below)` comment in the appropriate row in the able.